Author: Ferry Huberts <f.huberts@mindef.nl>
Date:   Tue Dec 6 16:51:10 2011 +0100
Description: align olsr_ip_addr to olsr definition of it
Forwarded: already included upstream
Last-Update: <2011-12-06>
Applied-Upstream: 8745ca2f91db8b0bb761e1380493b16b7a1c192a

diff --git a/gui/linux-gtk/src/ipc.c b/gui/linux-gtk/src/ipc.c
index c93646a..d1867e0 100644
--- a/gui/linux-gtk/src/ipc.c
+++ b/gui/linux-gtk/src/ipc.c
@@ -586,7 +586,7 @@ ip_to_string(union olsr_ip_addr *addr)
   struct in_addr in;
 
   if (ipversion == AF_INET) {
-    in.s_addr = addr->v4;
+    in = addr->v4;
     ret = inet_ntoa(in);
   } else {
     /* IPv6 */
diff --git a/gui/linux-gtk/src/olsr_protocol.h b/gui/linux-gtk/src/olsr_protocol.h
index a458a7c..ef084a5 100644
--- a/gui/linux-gtk/src/olsr_protocol.h
+++ b/gui/linux-gtk/src/olsr_protocol.h
@@ -60,7 +60,7 @@ typedef int32_t olsr_32_t;
 #include <netinet/in.h>
 
 union olsr_ip_addr {
-  olsr_u32_t v4;
+  struct in_addr v4;
   struct in6_addr v6;
 };
 
